test/framework/src/main/java/org/elasticsearch/common/settings/MockSecureSettings.java

```diff
@@ -36,10 +37,10 @@
  */
 public class MockSecureSettings implements SecureSettings {
 
-    private Map<String, String> secureStrings = new HashMap<>();
-    private Map<String, byte[]> files = new HashMap<>();
-    private Map<String, byte[]> sha256Digests = new HashMap<>();
-    private Set<String> settingNames = new HashSet<>();
+    private Map<String, String> secureStrings = Collections.synchronizedMap(new HashMap<>());
```

Comment:
Actually they are required.
In integ tests, I change values of secure settings, which are picked up by the threads of the cluster nodes. I need to assure that a change of secure settings is visible after I restart the cluster node.

I understand your reticence of changing the infra code, so I've created `SynchronizedMockSecureSettings#SynchronizedMockSecureSettings` which synchronizes access to every method, and reverted this changes.

Let me know if you prefer back these changes.

Reply:
I still think you don't need synchronisation here at all. In the one test that you're mutating the secure settings in, you are first setting the password, then starting the nodes? Even if on different threads, the order of operations is clear in memory and you shouldn't need synchronisation? 

I'm not so much worried about the change, I'm more worried that tests will start behaving unpredictably if we need this change. If we need it, it would imply that it's not completely clear whether or not we picked up one version or the other of the changed password setting?

Reply:
We've discussed this on another channel and Armin's right.

Restarting a node should obviously pick up all changes effectuated on the shared data by the test thread. I was under the impression that the node restart is like a "soft reset" where the node thread continues execution and simply reloads the configuration (No reason to assume something like that but I was convinced of it).